server/src/model.py

```diff
@@ -0,0 +1,62 @@
+import string
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.metrics import accuracy_score, classification_report
+from nltk.corpus import stopwords
+from sklearn.ensemble import RandomForestClassifier
+
+
+# approach:
+# clean and preprocess the questions (remove punctuation, lowercase the text, etc.).
+# remove stop words like "is", "my", "the"
+# use TF-IDF (Term Frequency-Inverse Document Frequency) to convert the questions into numerical vectors
+# in comparison to CountVectorizer, CV just counts the frequency of a word in a given sentence/input
+# TF_IDF counts the frequency AND the Inverse Document Frequency
+# use Random Forest Classifier 
+# predict the intent of the user's question/input
+
+import re
+def model(question: string): 
+    df = pd.read_excel('server/src/dataset/OIT_Dataset.xlsx', sheet_name="Sheet1")
+    df = df.sample(frac=1)
+
+    df['Question'] = df['Question'].str.lower().str.replace('[^\w\s]', '', regex=True)
+    # nltk.download('stopwords')
```

Comment:
Probably call this outside the function so that we don't attempt to download this everytime.

Reply:
which function?